tests/tests/bitasset_tests.cpp

```diff
@@ -446,57 +404,199 @@ BOOST_AUTO_TEST_CASE( reset_backing_asset_on_non_witness_asset )
    }
    {
       BOOST_TEST_MESSAGE("After hardfork, change underlying asset of bit_jmj from core to bit_usd");
+      change_backing_asset(*this, nathan_private_key, bit_jmj_id, bit_usd_id);
+
+      BOOST_TEST_MESSAGE("Verify feed producers have been reset");
+      const asset_bitasset_data_object& jmj_obj = bit_jmj_id(db).bitasset_data(db);
+      BOOST_CHECK_EQUAL(jmj_obj.feeds.size(), 3ul);
+      for(const auto& feed : jmj_obj.feeds) {
+         BOOST_CHECK(std::isnan(feed.second.second.settlement_price.to_real()));
+      }
+   }
+   {
+      BOOST_TEST_MESSAGE("With underlying bitasset changed from one to another, price feeds should still be publish-able");
+      BOOST_TEST_MESSAGE("Adding Vikram's price feed");
+      add_price_feed(*this, vikram_id, bit_usd_id, 1, bit_jmj_id, 30, core_id);
+
+      const asset_bitasset_data_object& bitasset = bit_jmj_id(db).bitasset_data(db);
+      BOOST_CHECK_EQUAL(bitasset.current_feed.settlement_price.to_real(), 30);
+      BOOST_CHECK(bitasset.current_feed.maintenance_collateral_ratio == GRAPHENE_DEFAULT_MAINTENANCE_COLLATERAL_RATIO);
+
+      BOOST_TEST_MESSAGE("Adding Ben's pricing to JMJBIT");
+      add_price_feed(*this, ben_id, bit_usd_id, 1, bit_jmj_id, 25, core_id);
+
+      BOOST_CHECK_EQUAL(bitasset.current_feed.settlement_price.to_real(), 30);
+      BOOST_CHECK(bitasset.current_feed.maintenance_collateral_ratio == GRAPHENE_DEFAULT_MAINTENANCE_COLLATERAL_RATIO);
+
+      BOOST_TEST_MESSAGE("Adding Dan's pricing to JMJBIT");
+      add_price_feed(*this, dan_id, bit_usd_id, 1, bit_jmj_id, 10, core_id);
+
+      BOOST_CHECK_EQUAL(bitasset.current_feed.settlement_price.to_real(), 25);
+      BOOST_CHECK(bitasset.current_feed.maintenance_collateral_ratio == GRAPHENE_DEFAULT_MAINTENANCE_COLLATERAL_RATIO);
+      generate_block();
+      trx.clear();
+
+      BOOST_CHECK(bitasset.current_feed.core_exchange_rate.base.asset_id != bitasset.current_feed.core_exchange_rate.quote.asset_id);
+   }
+}
+
+/*****
+ * @brief make sure feeds work correctly after changing from non-witness-fed to witness-fed before the 868 fork
+ */
```

Comment:
This test case belongs to a new issue, especially when it will fail.

Reply:
Temporarily commented out this test.